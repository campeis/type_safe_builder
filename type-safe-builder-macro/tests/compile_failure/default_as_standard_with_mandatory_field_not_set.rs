use type_safe_builder_macro::Builder;

#[derive(Builder)]
#[builder(default)]
struct StructToBuild {
    #[builder(mandatory)]
    field: i64,
}

fn main() {
    let _ = StructToBuildBuilder::builder().build();
}
