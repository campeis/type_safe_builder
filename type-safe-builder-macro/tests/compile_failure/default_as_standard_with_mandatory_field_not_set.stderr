error[E0599]: no method named `build` found for struct `StructToBuildBuilderState<false>` in the current scope
 --> tests/compile_failure/default_as_standard_with_mandatory_field_not_set.rs:11:45
  |
 3 | #[derive(Builder)]
   |          ------- method `build` not found for this struct
...
11 |     let _ = StructToBuildBuilder::builder().build();
   |                                             ^^^^^ method not found in `StructToBuildBuilderState<false>`
   |
   = note: the method was found for
           - `StructToBuildBuilderState<true>`
